models/marts/tests/intermediate/int_model_test_summary.sql

```diff
@@ -32,6 +38,34 @@ count_column_tests as (
     group by 1,2
 ),
 
+count_column_constraints as (
+
+    select
+        node_unique_id as direct_parent_id,
+        name as column_name,
+        case
+            when replace(constraints, 'not_null', '') != constraints
```

Comment:
not a huge deal, but worth calling out that this approach means we'd count anything that matches this string like `not_not_null` or `not_nulls` as a not_null constraint -- probably pretty unlikely given what kind of constraints exist out there in the wild, but worth calling out if someone uses the [expression option](https://docs.getdbt.com/reference/resource-properties/constraints) and does something related to not null we may get false positives.

Maybe we could check for `"type": "not_null"` or simply `"not_null"`?

Reply:
Good call! I've updated the line to check for `"not_null"`.